qiskit/algorithms/time_evolvers/classical_methods/utils.py

```diff
@@ -0,0 +1,181 @@
+
+    """
+
+    aux_ops = evolution_problem.aux_operators
+
+    time_array = np.linspace(0, evolution_problem.time, ops_ev_mean.shape[-1])
+    zero_array = np.zeros(ops_ev_mean.shape[-1])  # std=0 since it is an exact method
+
+    operators_number = 0 if aux_ops is None else len(aux_ops)
+
+    observable_evolution = [(ops_ev_mean[i], zero_array) for i in range(operators_number)]
+
+    if isinstance(aux_ops, dict):
+        observable_evolution = dict(zip(aux_ops.keys(), observable_evolution))
+        observable_evolution["time"] = time_array
+    else:
+        observable_evolution += [time_array]
+
+    return observable_evolution
+
+
+def _create_obs_final(
+    ops_ev_mean: np.ndarray,
+    evolution_problem: TimeEvolutionProblem,
+) -> ListOrDict[tuple[complex, complex]]:
+    """Creates the right output format for the final value of the auxiliary operators.
+
+    Args:
+        ops_ev_mean: Array containing the expectation value of each observable at the final timestep.
+        evolution_problem: Evolution problem to create the output of.
+
+    Returns:
+        An output with the observables mean value at the appropriate times depending on whether
+        the auxiliary operators in the evolution problem are a `list` or a `dict`.
+
+    """
+
+    aux_ops = evolution_problem.aux_operators
+    aux_ops_evaluated = [(op_ev, 0) for op_ev in ops_ev_mean]
+    if isinstance(aux_ops, dict):
+        aux_ops_evaluated = dict(zip(aux_ops.keys(), aux_ops_evaluated))
+    return aux_ops_evaluated
+
+
+def _evaluate_aux_ops(
+    aux_ops: list[csr_matrix],
+    state: np.ndarray,
+) -> tuple[np.ndarray, np.ndarray]:
+    """Evaluates the aux operators if they are provided and stores their value.
+
+    Returns:
+        Tuple of the mean and standard deviation of the aux operators for a given state.
+    """
+    op_means = np.array([np.real(state.conjugate().dot(op.dot(state))) for op in aux_ops])
+    return op_means
+
+
+def _sparsify(
+    evolution_problem: TimeEvolutionProblem, steps: int, real_time: bool
+) -> tuple[np.ndarray, list[csr_matrix], csr_matrix]:
+    """Returns the matrices and parameters needed for time evolution in the appropriate format.
+
+    Args:
+        evolution_problem: The definition of the evolution problem.
+        steps: Number of timesteps to be performed.
+        real_time: If `True`, returned operators will correspond to real time evolution,
+            Else, they will correspond to imaginary time evolution.
+
+    Returns:
+        A tuple with the initial state, the list of operators to evaluate and the operator to be
+        exponentiated to perform one timestep.
+    """
+    # Convert the initial state and Hamiltonian into sparse matrices.
+    if isinstance(evolution_problem.initial_state, QuantumCircuit):
+        state = Statevector(evolution_problem.initial_state).data.T
+    else:
+        state = evolution_problem.initial_state.to_matrix(massive=True).transpose()
```

Comment:
We changed from `StateFn` in the time evolution problem to `Statevector`, so we have to switch to `Statevector` support here 🙂